geetools/ee_image_collection.py

```diff
@@ -896,6 +896,25 @@ def groupInterval(self, unit: str = "month", duration: int = 1) -> ee.List:
             lambda dr: ic.filterDate(ee.DateRange(dr).start(), ee.DateRange(dr).end())
         )
 
+        sizeName = "__geetools_generated_size__"
+
+        def add_size(ic):
+            ic = ee.ImageCollection(ic)
+            return ic.set({sizeName: ic.size()})
+
+        def delete_size_property(ic):
+            ic = ee.ImageCollection(ic)
+            propertiesToCopy = ic.propertyNames().remove(sizeName)
+            return ee.ImageCollection(ic.toList(ic.size())).copyProperties(
+                ic, properties=propertiesToCopy
+            )
+
+        imageCollectionList = (
```

Comment:
I'm disturb by this block. I try to think of a way to achieve the same goal without the double mapping. @fitoprincipe  maybe you would have an idea ?